apps/studio/prisma/schema.prisma

```diff
@@ -46,16 +56,25 @@ model Resource {
 
   permission Permission[]
 
-  mainBlob   Blob?   @relation("MainBlob", fields: [mainBlobId], references: [id])
-  mainBlobId BigInt? @unique
+  versionId BigInt?
+  version   Version? @relation(fields: [versionId], references: [id])
 
-  draftBlob   Blob?   @relation("DraftBlob", fields: [draftBlobId], references: [id])
   draftBlobId BigInt? @unique
+  draftBlob   Blob?   @relation(fields: [draftBlobId], references: [id])
 
   state ResourceState? @default(Draft)
   type  ResourceType
 }
 
+model Blob {
+  id            BigInt    @id @default(autoincrement())
+  /// @kyselyType(PrismaJson.BlobJsonContent)
+  /// [BlobJsonContent]
+  content       Json
+  versions      Version[]
```

Comment:
thought: the way I see this is that a blob is a unique instance that exists for a single version (if published) or as a draft, so this might refer back to the specific version that the blob is tied to.